components/delay/delay.go

```diff
@@ -0,0 +1,65 @@
+package delay
+
+import (
+	"context"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// Delay represents a message's delay.
+// It can be either a delay until a specific time or a delay for a specific duration.
+// The zero value of Delay is a zero delay.
+//
+// IMPORTANT: Delay doesn't work with all Pub/Subs! Using it won't have any effect on Pub/Subs that don't support it.
```

Comment:
I wonder if we can somehow ensure that someone is not using Pub/Sub that doesn't support it 🤔 

as far as I understand, now you can use whatever pubsub and if it's not compatible it will "silently not work"?

I can imagine that if someone can accedantly inject wrong one, it won't be visible

it would be nice if we could do it compile time, or at least return some error so it won't be totally silent

Reply:
As discussed, this is not exactly possible, but I added the opposite: b7fb98bf9b08d4b31627eee3400368bfd262b2d4

Reply:
Added also `AllowNoDelay` to the delayed subscriber's config.